File: user-service/src/main/java/com/vb/springboot/user/service/UsersService.java
package com.vb.springboot.user.service;

import com.vb.springboot.user.shared.UserDto;
import com.vb.springboot.user.model.request.UpdateUserDetailsRequestModel;
import com.vb.springboot.user.model.response.UserRestResponse;

public interface UsersService {

    UserDto createUser(UserDto userDetails);

    UserRestResponse getUser(String userId);

    UserRestResponse updateUser(String userId, UpdateUserDetailsRequestModel userToUpdate);

    void deleteUser(String userId);
}

File: user-service/src/main/kotlin/com/vb/springboot/user/service/UsersService.kt
package com.vb.springboot.user.service

import com.vb.springboot.user.shared.UserDto

interface UsersService {
    fun createUser(userDetails: UserDto): UserDto

    fun getUser(userId: String): UserDto?

    fun updateUser(userId: String, userToUpdate: UserDto): UserDto?

    fun deleteUser(userId: String)
}